packages/plugins/response-cache/src/plugin.ts

```diff
@@ -168,6 +176,10 @@ export function useResponseCache({
       const identifier = new Map<string, CacheEntityRecord>();
       const types = new Set<string>();
 
+      if (!cacheIntrospections && isIntrospectionDocument(ctx.args.document)) {
```

Comment:
Maybe this could go to the beginning of the `onExecute` function? just to avoid creating the Map/Set objects

Reply:
This is so strange, I don't have the code in onParse anymore but in

```ts
    async onExecute(ctx) {
      const identifier = new Map<string, CacheEntityRecord>();
      const types = new Set<string>();

      if (!cacheIntrospections && isIntrospectionDocument(ctx.args.document)) {
        return;
      }
````

But, it is late and I may not be reading my code properly :)

Reply:
All logic for if should cache result is not in a customizable function: shouldCacheResult.